apps/kytrade/src/kytrade/analysis.py

```
@@ -0,0 +1,216 @@
+        universe=len(histories),
+    )
+
+
+def volatility(symbol: str, window_days: int = 21) -> VolatilityReport:
+    """Daily-return volatility for a symbol and its worst rolling window."""
+    points = _closes(stocks.get_prices(symbol))
+    if len(points) <= window_days:
+        raise NotEnoughData(f"{symbol}: need more than {window_days} stored closes")
+    returns = [
+        (day, (close / prev_close - 1) * 100)
+        for (_, prev_close), (day, close) in zip(points, points[1:], strict=False)
```

Ruff (0.15.20)

[warning] Prefer `itertools.pairwise()` over `zip()` when iterating over successive pairs

Replace `zip()` with `itertools.pairwise()`

(RUF007)